clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -79,6 +84,46 @@ public static boolean hasId(int id) {
         return id >= MIN_API_KEY && id <= MAX_API_KEY;
     }
 
+    public short latestVersion() {
+        if (id >= Protocol.CURR_VERSION.length)
```

Comment:
As @hachikuji said, we don't need to check the id in every method because the set of ids is statically defined in this class. I added a check in the constructor to be completely safe.

Regarding the `Protocol.CURR_VERSION` check, I think it's a useful debugging aid for people who update the protocol arrays and it's basically free in the JVM (it can elide the array bounds check when accessing the array).